mobile-app/app/screens/AppNavigator/screens/Balances/screens/AddressBookScreen.tsx

```diff
@@ -86,58 +84,40 @@ export function AddressBookScreen ({ route, navigation }: Props): JSX.Element {
   }, [addresses])
 
   useEffect(() => {
-    if (showSearchInput) {
-      filterAddress(searchString) // filter while searching
-    }
+    filterAddress(searchString) // filter while searching
   }, [searchString])
 
-  useLayoutEffect(() => {
-    navigation.setOptions({
-      headerRight: (): JSX.Element => {
-        // don't display search icon if there are no addresses
-        if (addresses.length > 0) {
-          return (
-            <HeaderSearchIcon
-              onPress={() => {
-                setShowSearchInput(true)
-                setFilteredAddresses([])
-              }}
-              testID='address_search_icon'
-            />
-          )
-        } else {
-          return <></>
-        }
-      }
-    })
-  }, [navigation, showSearchInput, addresses])
-
   useEffect(() => {
-    if (showSearchInput) {
-      navigation.setOptions({
-        header: (): JSX.Element => (
-          <HeaderSearchInput
-            searchString={searchString}
-            onClearInput={() => setSearchString('')}
-            onChangeInput={(text: string) => {
-              setSearchString(text)
-            }}
-            onCancelPress={() => {
-              setSearchString('')
-              setShowSearchInput(false)
-              setFilteredAddresses(addresses)
-            }}
-            placeholder={translate('screens/AddressBookScreen', 'Search for address')}
-            testID='address_search_input'
+    navigation.setOptions({
+      headerTitle: (): JSX.Element => (
+        <SearchInput
+          value={searchString}
+          placeholder={translate('screens/AddressBookScreen', 'Search for address')}
+          showClearButton={searchString !== ''}
+          onClearInput={() => setSearchString('')}
+          onChangeText={(text: string) => {
+            setSearchString(text)
+          }}
+          testID='address_search_input'
+        />
+      ),
+      headerRight: (): JSX.Element => (
+        <TouchableOpacity
+          onPress={goToAddAddressForm}
+          testID='add_address_button'
+        >
+          <ThemedIcon
+            size={28}
+            name='plus'
+            style={tailwind('mr-2')}
```

Comment:
Can we place the `+` icon slightly more towards the left? Also can we increase the space below the search bar in iOS?
![IMG_F59E8862489B-1](https://user-images.githubusercontent.com/40191153/170950779-d588a893-0ac5-4cde-975a-8f5fe4abc0f2.jpeg)